### PR TITLE
Devcontainer for python-threatexchange

### PR DESCRIPTION
**Pull request by schatten**
Summary
---------
In some distant future, we might be able to do a single devcontainer for both python-threatexchange and hma, but for now, have a separate one.

Test Plan
---------
Booted the devcontainer using "Cmd+P" -> "Remote Container: Open Folder.."

The necessary extensions were installed. Once I selected the python interpreter, I could cmd-click into library functions.

### CODE REVIEW

**Review by BarrettOlson — APPROVED** [review]
Approving to unblock though this does feel like a bit overkill (especially if both are given 4GB of memory). I would think something like a script to create a python `venv` would be enough for this directory (and maybe a .vscode workspace settings file).

If we do go this route we should have both containers use the same base layer to reduce the point of maintenance because I'm sure there will be drift. 

Thanks!

File: python-threatexchange/.devcontainer/zshrc
Comment (BarrettOlson) [inline python-threatexchange/.devcontainer/zshrc:17]: I think this needs to be updated...

Also could you have this also be added to `/home/$unixname/.profile` for those using https://code.visualstudio.com/docs/remote/containers#_personalizing-with-dotfile-repositories

File: python-threatexchange/.devcontainer/bashrc
Comment (BarrettOlson) [inline python-threatexchange/.devcontainer/bashrc:8]: same note from the zshrc file

**Review by BarrettOlson — COMMENTED** [review]
See note about PYTHONPATH before merge.
Also might be worth adding a post-create scripts that will do the`make local_install` step

File: python-threatexchange/.devcontainer/zshrc
Comment (schatten) [reply python-threatexchange/.devcontainer/zshrc:17]: Updating the path, but I'm not sure what .profile would do here. Feel free to move the files around. I only got things to a point where they work. I'm _certain_ I have not followed best practices. :)

Comment (BarrettOlson) [reply python-threatexchange/.devcontainer/zshrc]: I don't know that the best practice is but the way I use the `.profile` file is in my zshenv:
```
# for machine specific configuration
if [[ -f ~/.profile ]]; then
  source ~/.profile
fi
```
https://github.com/BarrettOlson/dotfiles/blob/master/zshenv#L31-L34

**Pull request merged**
